sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParser.java

```diff
@@ -108,17 +120,17 @@ private void handleSequencePointTag(XmlParserHelper xmlParserHelper) {
       }
 
       if (files.containsKey(fileId)) {
-        String filePath = files.get(fileId);
+        Optional<String> optionalPath = files.get(fileId);
 
-        if (fileService.isSupportedAbsolute(filePath)) {
-          LOG.trace("OpenCover parser: add hits for fileId '{}', filePath '{}', line '{}', visitCount '{}'",
-            fileId, filePath, line, visitCount);
+        if (optionalPath.isPresent()) {
+          String absoluteFilePath = optionalPath.get();
```

Comment:
I think the name can be misleading since, at  this point, we don't know for sure if the path is absolute or relative. We can name it just `filePath` or `path`.

Reply:
I thought that a functional approach with `ifPresentOrElse` would work better but it seems that we need to set the language version to 9 in order to be able to do that.

Reply:
good point, I should update to Java 11 which is the current LTS